libindy_vdr/src/pool/cache/mod.rs

```diff
@@ -0,0 +1,61 @@
+use async_lock::RwLock;
+use async_trait::async_trait;
+use std::{fmt::Display, sync::Arc};
+
+pub mod storage;
+pub mod strategy;
+
+#[async_trait]
+pub trait CacheStrategy<K, V>: Send + Sync + 'static {
+    async fn get(&self, key: &K) -> Option<V>;
+
+    async fn remove(&mut self, key: &K) -> Option<V>;
+
+    async fn insert(&mut self, key: K, value: V, custom_exp_offset: Option<u128>) -> Option<V>;
+}
+
+pub struct Cache<K, V> {
+    storage: Arc<RwLock<dyn CacheStrategy<String, V>>>,
+    key_prefix: Option<K>,
+}
+
+impl<K: Display + 'static, V: 'static> Cache<K, V> {
+    fn full_key(&self, key: &K) -> String {
+        match &self.key_prefix {
+            Some(prefix) => format!("{}{}", prefix, key),
+            None => key.to_string(),
+        }
+    }
```

Comment:
I think rustfmt needs to be run on this file and the next? It seems odd not having a blank line between methods.

Reply:
rust fmt didn't seem to complain, I added the linebreaks anyway though